### ISSUE TITLE
Higher degree polynomials seem to produce outliers in implicit function evaluation

### ISSUE THREAD

**Issue by oveddan** [danielepanozzo/gp-old#8]
When I am using 0 degrees for the ponymial, the mesh always generates nicely:

<img width="808" alt="screen shot 2017-03-13 at 11 19 35 am" src="https://cloud.githubusercontent.com/assets/891755/23869177/52cad84e-07df-11e7-886b-5560e3a73e32.png">

However, on degree 1 or 2, the mesh seems to be smoother, but there are a lot more outliers as far as negative implicit values outside of where the mesh should be, resulting in parts of the mesh being generated outside:

## With degree 1:
<img width="829" alt="screen shot 2017-03-13 at 11 20 31 am" src="https://cloud.githubusercontent.com/assets/891755/23869242/7e3b4c0c-07df-11e7-8b78-202c373f974e.png">
<img width="892" alt="screen shot 2017-03-13 at 11 19 54 am" src="https://cloud.githubusercontent.com/assets/891755/23869249/8311a064-07df-11e7-95d6-d2ebbb7b1cbe.png">

## With degree 2:
<img width="905" alt="screen shot 2017-03-13 at 11 20 21 am" src="https://cloud.githubusercontent.com/assets/891755/23869269/8ef76e04-07df-11e7-87d8-24da2d5678c6.png">
<img width="859" alt="screen shot 2017-03-13 at 11 20 05 am" src="https://cloud.githubusercontent.com/assets/891755/23869273/938f8b9a-07df-11e7-856e-4a57aeb64085.png">

Is this a common problem?  Any suggestions for tracking down what is causing this problem?

**Comment by danielepanozzo** [comment]
Yes this is common. As you increase the degree of the polynomial, you get more degrees of freedom and the scalar function is more likely to oscillate. To filter the noise, detect the groups of vertices connected by edges, and delete all the connected components except the larger one, which will correspond to the mesh. Increasing the resolution of the grid should also help.
